dshl/scope.go

```diff
@@ -0,0 +1,74 @@
+package dshl
+
+// A Scope represents a scope in the shell, holding variables and commands.
+// Scopes can be stacked, in a reverse linked list structure, and shadow each other.
+type Scope struct {
+	Parent *Scope // Parent scope.
+	Modal  bool   // A modal scope can be exitted with Ctrl+D.
+	PS1    PS1    // PS1 associated with this scope; can be shadowed.
+
+	vars map[string]interface{}
+}
+
+// Returns a child scope, with this scope set as its parent.
+func (s *Scope) Scope() *Scope {
```

Comment:
Would this not be better called `Child()`?

Reply:
Changed